File: dart/dart-sample/sample/oop/enum.dart
void main(List<String> args) {
  var trafficLight = TrafficLight.red;
  print(trafficLight);
  print(trafficLight.turn());

  for (var e in TrafficLight.values) {
    print(e.name);
  }
}

enum TrafficLight {
  red,
  yellow,
  green;

  TrafficLight turn() =>
      switch (this) { red => yellow, yellow => green, green => red };
}

File: scala/scala-example/src/main/scala/tech/techstay/oop/Enum.scala
package tech.techstay.oop

enum TrafficLight:
  def turn(): TrafficLight = this match
    case RED    => GREEN
    case GREEN  => YELLOW
    case YELLOW => RED
  case RED, GREEN, YELLOW

@main def enums(): Unit =
  var trafficLight = TrafficLight.RED
  println(trafficLight)
  println(trafficLight.turn())
  println(trafficLight.turn())
